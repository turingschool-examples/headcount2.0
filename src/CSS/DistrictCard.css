.year-red {
  color: red;
}

.year-black {
  color: green;
}

.district-card {
  text-align: center;
  box-shadow: 2px 2px 2px 2px #ccc;
}

.district-location {
  font-size: 1.5rem;
}
