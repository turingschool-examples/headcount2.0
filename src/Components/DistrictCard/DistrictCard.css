.highStats {
  color: magenta;
}

.lowStats {
  color: chartreuse;
}

.card {
  border-radius: 18px;
  border: solid black 2px;
  display: inline-block;
  margin: 1rem;
  padding: .5rem;
  width: 300px;

}

.selected {
  background-color: green;
  border: solid black 4px;
}

.card:hover {
  border: solid black 4px;
}

.district {
  font-family: avenir, sans-serif;
  font-size: 1.5em;
  text-align: center;
}

.stats {
  font-family: avenir, sans-serif;
  font-size: 1.25em;
  padding-left: 5rem;
}