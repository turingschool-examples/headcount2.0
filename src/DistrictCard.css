.districtCard {
  box-shadow: 1em;
  border-radius: 15px;
  background-color: white;
  display: flex;
  flex-direction: column;
  width: 15rem;
  height: auto;
  margin: 0 0 5rem 0;
  padding-bottom: 1rem;
  text-align: center
}

.districtLocation {
  font-size: 1rem
}

.districtStats {
  list-style-type: none;
  text-align: center;
  margin: 0;
  padding: 0
}

.statGreater {
  color: blue
}

.statLesser {
  color: red
}