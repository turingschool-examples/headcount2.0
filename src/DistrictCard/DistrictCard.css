.DistrictCard {
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: center;
	background-color: #FFF;
	text-align: center;
	width: 15rem;
	height: 26rem;
	margin: 0 auto 0.5rem;
	padding: 0 0 1rem;
	box-shadow: 0px 1px 1px 2px #F5F6F7;
	border-radius: 10px;
	font-family: 'Open Sans', sans-serif;
}

.card-location {
	text-align: center;
	overflow-wrap: break-word;
	margin: 1.5rem 1.5rem 0.75rem;
	font-family: 'Arvo', serif;
	font-size: 1.25rem;
}

/*.card-stats {
	margin: 0 auto 1.5rem;
}
*/
.greater-than-point-5 {
	color: #01411b;
	background-color: #F5F6F7;
	margin: 0.2rem auto;
	font-weight: 300;
	padding: 0 0.75rem;
	border-radius: 10px;
}

.less-than-point-5 {
	color: #FFF;
	background-color: #008b38;
	margin: 0.2rem auto;
	font-weight: 300;
	padding: 0 0.75rem;
	border-radius: 10px;
}


