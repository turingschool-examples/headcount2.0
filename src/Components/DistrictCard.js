import React from "react";
import "../CSS/DistrictCard.css";
import PropTypes from "prop-types";

export const DistrictCard = ({
  location,
  stats,
  clicked,
  selectCard,
  counter
}) => {
  return (
    <div
      onClick={() => selectCard(location)}
      className={clicked ? "district-card-clicked" + counter : "district-card"}
    >
      <h1 className="district-location">{location}</h1>
      <header className={clicked ? "year-header-show" : "year-header-hide"}>
        Year
      </header>
      <header className={clicked ? "avg-header-show" : "avg-header-hide"}>
        Avg
      </header>
      {Object.keys(stats).map((year, index) => {
        const avg = stats[year];
        return (
          <section
            className={clicked ? "stats-show" : "stats-hide"}
            key={index}
          >
            <p className="year">
              {year}
              ......
              <span className={avg < 0.5 ? "avg-low" : "avg-high"}>{avg}</span>
            </p>
          </section>
        );
      })}
      <p className={clicked ? "show-msg" : "hide-msg"}>Click to Remove</p>
    </div>
  );
};

const { string, shape, number, bool, func } = PropTypes;

DistrictCard.propTypes = {
  location: string,
  stats: shape({ year: number }),
  clicked: bool,
  selectCard: func,
  counter: number
};
