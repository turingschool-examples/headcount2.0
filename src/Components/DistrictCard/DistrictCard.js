import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';


const DistrictCard = ({district , selectDistrict}) => {
  const years = Object.keys(district.stats)

  const determineColor = (stats) => {
    return stats > 0.5 ? "highStats" : "lowStats"
  }

  const createStats = years.map((year, index) => {
    const stats = district.stats[year]
    return (
      <li className={determineColor(stats)} key={index}>
        {[year]}: {stats}
      </li>
      )
  })

  const handleClick = (location) => {
    selectDistrict(location)
  }

  return (
    <section 
      className={`card ${district.selected ? 'selected' : ''}`} 
      id={district.key}
      key={district.location}
      onClick={() => {handleClick(district.location)}}
    > 
      <h2 className="district">{district.location}</h2>
      <ul className="stats">
        {createStats}
      </ul>
    </section>
  )
}

DistrictCard.propTypes = { 
  district: PropTypes.object
}



export default DistrictCard;