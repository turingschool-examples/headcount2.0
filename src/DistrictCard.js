import React from "react";
import "./CSS/DistrictCard.css";

export const DistrictCard = ({ location, stats, selectCard }) => {
  return (
    <div className="district-card">
      <h1 className="district-location">{location}</h1>
      {Object.keys(stats).map((year, i) => {
        const avg = stats[year];
        return (
          <section
            className={avg < 0.5 ? "year-red" : "year-black"}
            key={i}
            onClick={e => selectCard(location)}
          >
            {year} :
            <span className={avg < 0.5 ? "year-red" : "year-black"}>{avg}</span>
          </section>
        );
      })}
    </div>
  );
};
