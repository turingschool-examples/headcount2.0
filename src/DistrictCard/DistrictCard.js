import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';

const DistrictCard = ({ location, stats, district, compareDistrict }) => {

	const statsKeys = Object.keys(stats)
	const schoolData = statsKeys.map((stat) => {
		// console.log(stat)
		return <p 
			key={Math.random()}
			className={(stats[stat] > 0.5) ? 'greater-than-point-5' : 'less-than-point-5'}>
			{stat}: {stats[stat]} 
		</p>
	})
	
if (compareDistrict) {
		return (
		<div className='DistrictCard' onClick={ () => compareDistrict(district) }>
			<h3 className='card-location'>{location}</h3>
			{schoolData}
		</div>
	)
	} else {
		return (
			<div className='DistrictCard'>
				<h3 className='card-location'>{location}</h3>
				{schoolData}
			</div>
		)
	}

}

DistrictCard.proptypes = {
	location: PropTypes.string.isRequired,
	stats: PropTypes.object.isRequired,
	district: PropTypes.object.isRequired,
	compareDistrict: PropTypes.func
}

export default DistrictCard;
