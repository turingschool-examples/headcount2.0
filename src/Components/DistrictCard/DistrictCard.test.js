import React from 'react';
import { shallow } from 'enzyme';
import DistrictCard from './DistrictCard';


describe('DistrictCard unit test', () => {
  let wrapper;
  let mockDistrict; 
  let mockHandleClick

  beforeEach(() => {
    mockDistrict =  { stats: 
                              { 2004: 0.24,
                                2005: 0.278,
                                2006: 0.337,
                                2007: 0.395,
                                2008: 0.536,
                                2009: 0.598,
                                2010: 0.64,
                                2011: 0.672,
                                2012: 0.695,
                                2013: 0.703,
                                2014: 0.741 },
                             location: 'COLORADO' }
    mockHandleClick = jest.fn();
    wrapper = shallow(<DistrictCard district={mockDistrict} selectDistrict={mockHandleClick}/>);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
    });

  it('calls handleClick with the correct location ', () => {
    const expectedLocation = 'COLORADO' 
    const card = wrapper.find('.card')

    card.simulate('click')

    expect(mockHandleClick).toHaveBeenCalledWith(expectedLocation)
    });

  });
