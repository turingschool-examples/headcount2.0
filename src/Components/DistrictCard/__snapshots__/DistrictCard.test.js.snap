// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictCard unit test should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictCard
    district={
        Object {
            "location": "COLORADO",
            "stats": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
          }
    }
    selectDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "COLORADO",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="district"
>
          COLORADO
</h2>,
        <ul
          className="stats"
>
          <li
                    className="lowStats"
          >
                    2004
                    : 
                    0.24
          </li>
          <li
                    className="lowStats"
          >
                    2005
                    : 
                    0.278
          </li>
          <li
                    className="lowStats"
          >
                    2006
                    : 
                    0.337
          </li>
          <li
                    className="lowStats"
          >
                    2007
                    : 
                    0.395
          </li>
          <li
                    className="highStats"
          >
                    2008
                    : 
                    0.536
          </li>
          <li
                    className="highStats"
          >
                    2009
                    : 
                    0.598
          </li>
          <li
                    className="highStats"
          >
                    2010
                    : 
                    0.64
          </li>
          <li
                    className="highStats"
          >
                    2011
                    : 
                    0.672
          </li>
          <li
                    className="highStats"
          >
                    2012
                    : 
                    0.695
          </li>
          <li
                    className="highStats"
          >
                    2013
                    : 
                    0.703
          </li>
          <li
                    className="highStats"
          >
                    2014
                    : 
                    0.741
          </li>
</ul>,
      ],
      "className": "card ",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
          "className": "district",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="lowStats"
>
              2004
              : 
              0.24
</li>,
            <li
              className="lowStats"
>
              2005
              : 
              0.278
</li>,
            <li
              className="lowStats"
>
              2006
              : 
              0.337
</li>,
            <li
              className="lowStats"
>
              2007
              : 
              0.395
</li>,
            <li
              className="highStats"
>
              2008
              : 
              0.536
</li>,
            <li
              className="highStats"
>
              2009
              : 
              0.598
</li>,
            <li
              className="highStats"
>
              2010
              : 
              0.64
</li>,
            <li
              className="highStats"
>
              2011
              : 
              0.672
</li>,
            <li
              className="highStats"
>
              2012
              : 
              0.695
</li>,
            <li
              className="highStats"
>
              2013
              : 
              0.703
</li>,
            <li
              className="highStats"
>
              2014
              : 
              0.741
</li>,
          ],
          "className": "stats",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2004",
                ],
                ": ",
                0.24,
              ],
              "className": "lowStats",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.24,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2005",
                ],
                ": ",
                0.278,
              ],
              "className": "lowStats",
            },
            "ref": null,
            "rendered": Array [
              "2005",
              ": ",
              0.278,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2006",
                ],
                ": ",
                0.337,
              ],
              "className": "lowStats",
            },
            "ref": null,
            "rendered": Array [
              "2006",
              ": ",
              0.337,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2007",
                ],
                ": ",
                0.395,
              ],
              "className": "lowStats",
            },
            "ref": null,
            "rendered": Array [
              "2007",
              ": ",
              0.395,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2008",
                ],
                ": ",
                0.536,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2008",
              ": ",
              0.536,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2009",
                ],
                ": ",
                0.598,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2009",
              ": ",
              0.598,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2010",
                ],
                ": ",
                0.64,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2010",
              ": ",
              0.64,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2011",
                ],
                ": ",
                0.672,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2011",
              ": ",
              0.672,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2012",
                ],
                ": ",
                0.695,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2012",
              ": ",
              0.695,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2013",
                ],
                ": ",
                0.703,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2013",
              ": ",
              0.703,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  "2014",
                ],
                ": ",
                0.741,
              ],
              "className": "highStats",
            },
            "ref": null,
            "rendered": Array [
              "2014",
              ": ",
              0.741,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "COLORADO",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="district"
>
            COLORADO
</h2>,
          <ul
            className="stats"
>
            <li
                        className="lowStats"
            >
                        2004
                        : 
                        0.24
            </li>
            <li
                        className="lowStats"
            >
                        2005
                        : 
                        0.278
            </li>
            <li
                        className="lowStats"
            >
                        2006
                        : 
                        0.337
            </li>
            <li
                        className="lowStats"
            >
                        2007
                        : 
                        0.395
            </li>
            <li
                        className="highStats"
            >
                        2008
                        : 
                        0.536
            </li>
            <li
                        className="highStats"
            >
                        2009
                        : 
                        0.598
            </li>
            <li
                        className="highStats"
            >
                        2010
                        : 
                        0.64
            </li>
            <li
                        className="highStats"
            >
                        2011
                        : 
                        0.672
            </li>
            <li
                        className="highStats"
            >
                        2012
                        : 
                        0.695
            </li>
            <li
                        className="highStats"
            >
                        2013
                        : 
                        0.703
            </li>
            <li
                        className="highStats"
            >
                        2014
                        : 
                        0.741
            </li>
</ul>,
        ],
        "className": "card ",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
            "className": "district",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="lowStats"
>
                2004
                : 
                0.24
</li>,
              <li
                className="lowStats"
>
                2005
                : 
                0.278
</li>,
              <li
                className="lowStats"
>
                2006
                : 
                0.337
</li>,
              <li
                className="lowStats"
>
                2007
                : 
                0.395
</li>,
              <li
                className="highStats"
>
                2008
                : 
                0.536
</li>,
              <li
                className="highStats"
>
                2009
                : 
                0.598
</li>,
              <li
                className="highStats"
>
                2010
                : 
                0.64
</li>,
              <li
                className="highStats"
>
                2011
                : 
                0.672
</li>,
              <li
                className="highStats"
>
                2012
                : 
                0.695
</li>,
              <li
                className="highStats"
>
                2013
                : 
                0.703
</li>,
              <li
                className="highStats"
>
                2014
                : 
                0.741
</li>,
            ],
            "className": "stats",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2004",
                  ],
                  ": ",
                  0.24,
                ],
                "className": "lowStats",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                0.24,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2005",
                  ],
                  ": ",
                  0.278,
                ],
                "className": "lowStats",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                ": ",
                0.278,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2006",
                  ],
                  ": ",
                  0.337,
                ],
                "className": "lowStats",
              },
              "ref": null,
              "rendered": Array [
                "2006",
                ": ",
                0.337,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2007",
                  ],
                  ": ",
                  0.395,
                ],
                "className": "lowStats",
              },
              "ref": null,
              "rendered": Array [
                "2007",
                ": ",
                0.395,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2008",
                  ],
                  ": ",
                  0.536,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2008",
                ": ",
                0.536,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2009",
                  ],
                  ": ",
                  0.598,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2009",
                ": ",
                0.598,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2010",
                  ],
                  ": ",
                  0.64,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2010",
                ": ",
                0.64,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2011",
                  ],
                  ": ",
                  0.672,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2011",
                ": ",
                0.672,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2012",
                  ],
                  ": ",
                  0.695,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2012",
                ": ",
                0.695,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "9",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2013",
                  ],
                  ": ",
                  0.703,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2013",
                ": ",
                0.703,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    "2014",
                  ],
                  ": ",
                  0.741,
                ],
                "className": "highStats",
              },
              "ref": null,
              "rendered": Array [
                "2014",
                ": ",
                0.741,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
